### ISSUE TITLE
[`0.0.18`] `memory_efficient_attention` NaNs when `seqlen>32768`

### ISSUE THREAD

**Issue by comfyanonymous** [facebookresearch/xformers#719]
# 🐛 Bug

## To Reproduce

```
import xformers
import xformers.ops
import torch

q = torch.zeros(([1, 33728, 512])).cuda()
k = torch.zeros(([1, 33728, 512])).cuda()
v = torch.zeros(([1, 33728, 512])).cuda()
out = xformers.ops.memory_efficient_attention(q, k, v, attn_bias=None)
print(out)
print(torch.isnan(out).any())
```

## Expected behavior

out should not contain Nan.

## Environment

This test was done on the free google colab with their T4 GPU using the 0.0.18 package on pip.

## Additional context

It works fine on 0.0.17 but fails on 0.0.18.

People have been reporting that my ComfyUI returns black images during the VAE decoding phase when the resolution is higher than a certain amount and I have narrowed it down to this issue.

**Comment by leppie** [comment]
I hit this issue too yesterday after upgrading to 0.0.18.

**Comment by danthe3rd** [comment]
Hi,
Thanks for the minimal reproduction example. I can repro on A100 - I'll have a look

**Comment by danthe3rd** [comment]
It seems to happen due to a cast to int16 at some point in the code, so it happens when the sequence length is larger than `32768`. Updating the title according.

**Comment by danthe3rd** [comment]
I have a tentative fix - hopefully we can land that soon and release the `0.0.19` early next week to address that

**Comment by danthe3rd** [comment]
It should be fixed as of https://github.com/facebookresearch/xformers/commit/540fcbfdaf8230fd37644345a310a36af86c9d6b, and will be included in the next release (**0.0.19**). In the meantime, you can also use a development build `>=0.0.19.dev516`

**Comment by patrickvonplaten** [comment]
Thanks a mille for the fix @danthe3rd - very helpful thread here!
